modules/oadp-api-details.adoc

```diff
@@ -0,0 +1,240 @@
+
+|`noDefaultBackupLocation`
+|link:https://pkg.go.dev/builtin#bool[bool]
+|To install Velero without a default backup storage location, you must set the `noDefaultBackupLocation` flag in order to confirm installation.
+
+|`podConfig`
+|*link:https://pkg.go.dev/github.com/openshift/oadp-operator/api/v1alpha1#PodConfig[`PodConfig`]
+|Defines the configuration of the `Velero` pod.
+
+|`logLevel`
+|link:https://pkg.go.dev/builtin#string[string]
+|Velero server’s log level (use `debug` for the most granular logging, leave unset for Velero default). Valid options are `trace`, `debug`, `info`, `warning`, `error`, `fatal`, and `panic`.
+|===
+
+link:https://pkg.go.dev/github.com/openshift/oadp-operator/api/v1alpha1#VeleroConfig[Complete schema definitions for the type `VeleroConfig`].
+
+.CustomPlugin
+[options="header"]
+|===
+|Property|Type|Description
+
+|`name`
+|link:https://pkg.go.dev/builtin#string[string]
+|Name of custom plug-in.
+
+|`image`
+|link:https://pkg.go.dev/builtin#string[string]
+|Image of custom plug-in.
+|===
+
+link:https://pkg.go.dev/github.com/openshift/oadp-operator/api/v1alpha1#CustomPlugin[Complete schema definitions for the type `CustomPlugin`].
+
+.ResticConfig
+[options="header"]
+|===
+|Property|Type|Description
+
+|`enable`
+|*link:https://pkg.go.dev/builtin#bool[bool]
+|If set to `true`, enables backup and restore using Restic. If set to `false`, snapshots are needed.
+
+|`supplementalGroups`
+|[]link:https://pkg.go.dev/builtin#int64[int64]
+|Defines the Linux groups to be applied to the `Restic` pod.
+
+|`timeout`
+|link:https://pkg.go.dev/builtin#string[string]
+|A user-supplied duration string that defines the Restic timeout. Default value is `1hr` (1 hour). A duration string is a possibly signed sequence of decimal numbers, each with optional fraction and a unit suffix, such as `300ms`, -1.5h` or `2h45m`. Valid time units are `ns`, `us` (or `µs`), `ms`, `s`, `m`, and `h`.
+
+|`podConfig`
+|*link:https://pkg.go.dev/github.com/openshift/oadp-operator/api/v1alpha1#PodConfig[`PodConfig`]
+|Defines the configuration of the `Restic` pod.
+|===
+
+link:https://pkg.go.dev/github.com/openshift/oadp-operator/api/v1alpha1#ResticConfig[Complete schema definitions for the type `ResticConfig`].
+
+.PodConfig
+[options="header"]
+|===
+|Property|Type|Description
+
+|`nodeSelector`
+|map [ link:https://pkg.go.dev/builtin#string[string] ] link:https://pkg.go.dev/builtin#string[string]
+|Defines the `nodeSelector` to be supplied to a `Velero` `podSpec` or a `Restic` `podSpec`.
+
+|`tolerations`
+|[]link:https://pkg.go.dev/k8s.io/api/core/v1#Toleration[ Corev1.Toleration]
+|Defines the list of tolerations to be applied to a Velero deployment or a Restic `daemonset`.
+
+|`resourceAllocations`
+|link:https://pkg.go.dev/k8s.io/api/core/v1#ResourceRequirements[corev1.ResourceRequirements]
```

Comment:
maybe change the link's text to just  `ResourceRequirements` instead of corev1.ResourceRequirements as corev1 is not an actual package (it's an import alias). The package is actually v1, but I think it's redundant to mention it.